services/motion/builtin/move_request.go

```diff
@@ -214,6 +302,34 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		}
 	}
 
+	var obstacleDetectors []map[resource.Name]vision.Service
+	for _, obstacleDetectorNamePair := range motionCfg.ObstacleDetectors {
+		pair := make(map[resource.Name]vision.Service)
+		visionServiceName := obstacleDetectorNamePair.VisionServiceName
+		visionSvc, ok := ms.visionServices[visionServiceName]
+		if !ok {
+			return nil, resource.DependencyNotFoundError(visionServiceName)
+		}
+		cameraName := obstacleDetectorNamePair.CameraName
+		// make sure the specified pairs are valid
+		switch {
+		case visionServiceName.ContainsRemoteNames() && cameraName.ContainsRemoteNames():
```

Comment:
Is there not another way of determining compatibility between vision service and camera?  This approach feels a little error prone.  I would expect that the vision service exposes its dependent cameras in some way and that we could simply do a check to see if the camera is a dependent of the vision service before pairing them.